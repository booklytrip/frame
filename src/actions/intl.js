/**
 * @flow
 */

import type { Locale } from '../types';

/**
 * Change current locale to specified
 *
 * @param {String} locale - A new locale to set
 */
export const updateLocale = (locale: Locale) => ({
    type: 'INTL.UPDATE_LOCALE',
    locale,
});
