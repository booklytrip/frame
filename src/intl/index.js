import { addLocaleData } from 'react-intl';
import { reduce, mapValues } from 'lodash';

import en from 'react-intl/locale-data/en';
import lv from 'react-intl/locale-data/lv';
import ru from 'react-intl/locale-data/ru';

import * as defaultLocales from './locales';
import { locales as uikitLocales } from 'uikit';

addLocaleData([
    ...en,
    ...lv,
    ...ru,
]);

/**
 * Combine all messages from list of provides locales
 * and return single object
 */
const combineLocales = (locales) => (
    reduce(locales, (result, locale) => (
        mapValues(locale, (messages, language) => (
            { ...result[language], ...messages }
        ))
    ), {})
);

const locales = combineLocales([defaultLocales, uikitLocales]);

export { locales };
