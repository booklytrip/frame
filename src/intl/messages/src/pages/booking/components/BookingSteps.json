[
  {
    "id": "booking.BookingSteps.step1",
    "defaultMessage": "Passenger information"
  },
  {
    "id": "booking.BookingSteps.step2",
    "defaultMessage": "Choose your payment"
  },
  {
    "id": "booking.BookingSteps.step3",
    "defaultMessage": "Booking confirmation"
  }
]