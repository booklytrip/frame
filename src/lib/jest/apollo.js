import React from 'react';

import { ApolloProvider as Provider } from 'react-apollo';
import ApolloClient from 'apollo-client';
const client = new ApolloClient();

export const ApolloProvider = ({ children }) => (
    <Provider client={ client }>
        { children }
    </Provider>
);
