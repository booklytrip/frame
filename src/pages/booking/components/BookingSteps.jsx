/**
 * @flow
 */

import React from 'react';
import { injectIntl, defineMessages } from 'react-intl';
import styled from 'styled-components';

import { Grid, Label } from 'semantic-ui-react';

import type { Intl } from '../../../types';

type BookingStepsProps = {
    active: number,
    intl: Intl,
};

type StepProps = {
    text: string,
    badge: number,
    active?: boolean,
};

const intlMessages = defineMessages({
    step1: {
        id: 'booking.BookingSteps.step1',
        defaultMessage: 'Passenger information',
    },
    step2: {
        id: 'booking.BookingSteps.step2',
        defaultMessage: 'Choose your payment',
    },
    step3: {
        id: 'booking.BookingSteps.step3',
        defaultMessage: 'Booking confirmation',
    },
});

const HorizontalLine = styled.div`
    content: '';
    height: 2px;
    width: 60%;
    border-bottom: 1px solid #d4d4d4;
    position: absolute;
    display: block;
    margin-top: -2px;
    right: 13px;
    top: 50%;
    z-index: 1;
`;

const Badge = styled(Label)`
    &.ui.label {
        width: 35px;
        height: 35px;
        padding: 10px;
        font-size: 14px;
        font-weight: 600;
        line-height: 14px;
        text-align: center;
        vertical-align: middle;
        margin-right: 10px;
        border-radius: 500rem;
        color: ${({ theme }) => theme.booking.stepsBadge.color} !important;
        border-color: ${({ theme }) =>
            theme.booking.stepsBadge.borderColor} !important;
    }
`;

const StepText = styled.span`
    background-color: #ffffff;
    z-index: 2;
    padding-left: 5px;
    padding-right: 10px;
    position: relative;
    font-size: 16px;
    font-weight: bold;
    ${props =>
        props.active && `color: ${props.theme.booking.activeStepText.color};`};
`;

const Step = ({ text, badge, active }: StepProps) =>
    <Grid.Column>
        <HorizontalLine />
        <Badge basic>
            {badge}
        </Badge>
        <StepText active={active}>
            {text}
        </StepText>
    </Grid.Column>;

Step.defaultProps = {
    active: false,
};

const BookingSteps = injectIntl(({ active, intl }: BookingStepsProps) =>
    <Grid columns={3}>
        <Step
            text={intl.formatMessage(intlMessages.step1)}
            badge={1}
            active={active === 1}
        />
        <Step
            text={intl.formatMessage(intlMessages.step2)}
            badge={2}
            active={active === 2}
        />
        <Step
            text={intl.formatMessage(intlMessages.step3)}
            badge={3}
            active={active === 3}
        />
    </Grid>,
);

BookingSteps.defaultProps = {
    active: 1,
};

export default BookingSteps;
