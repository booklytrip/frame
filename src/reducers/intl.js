/**
 * @flow
 */

import { Map } from 'immutable';
import { saveState, loadState } from '../lib/sessionStorage';

const initialState = new Map(loadState('intl') || {});

export default function(state: Object = initialState, action: Object) {
    switch (action.type) {
        // Replace current locale with specified
        case 'INTL.UPDATE_LOCALE': {
            const newState = state.set('locale', action.locale);
            saveState('intl', newState);
            return newState;
        }
        default:
            return state;
    }
}
