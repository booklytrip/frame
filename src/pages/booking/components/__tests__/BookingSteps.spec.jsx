import React from 'react';
import { mountWithIntl, createWithIntl } from '../../../../lib/jest/intl';
import BookingSteps from '../BookingSteps';

describe('BookingSteps', () => {
    it('renders withut crashing', () => {
        mountWithIntl(<BookingSteps />);
    });

    describe('Snapshots', () => {
        it('renders with active step', () => {
            const rendered = createWithIntl(
                <BookingSteps active={ 2 } />
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    })
});
