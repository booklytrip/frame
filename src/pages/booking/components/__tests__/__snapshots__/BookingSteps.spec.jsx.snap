// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingSteps Snapshots renders with active step 1`] = `
<div
  className="ui three column grid"
>
  <div
    className="column"
  >
    <div
      className="hYpro"
    />
    <div
      className="ui blue basic label gNphye"
      onClick={[Function]}
    >
      1
    </div>
    <span
      className="dOjtQQ"
    >
      Passenger information
    </span>
  </div>
  <div
    className="column"
  >
    <div
      className="hYpro"
    />
    <div
      className="ui blue basic label gNphye"
      onClick={[Function]}
    >
      2
    </div>
    <span
      className="irAxMC"
    >
      Choose your payment
    </span>
  </div>
  <div
    className="column"
  >
    <div
      className="hYpro"
    />
    <div
      className="ui blue basic label gNphye"
      onClick={[Function]}
    >
      3
    </div>
    <span
      className="dOjtQQ"
    >
      Booking confirmation
    </span>
  </div>
</div>
`;
